WPMedia/fusion-news-theme-blocks
# Pen 1547 reduce hooks linting

## Description
Ensure rule of hooks issues addressed, no linting errors

## Jira Ticket
- [PEN-1547](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=545&projectKey=PEN&modal=detail&selectedIssue=PEN-1547)

## Acceptance Criteria
- no linting errors. exhaustive deps is an error not warn

## Test Steps
- Run `npm run storybook`
-Ensure search nav box does not allow you to submit while transitioning open
- no doubleclick issues
- clicking away should still close the input 
- npm run lint
- npx fusion start -f -l @wpmedia/ads-block,@wpmedia/header-nav-block,@wpmedia/header-nav-chain-block

## Effect Of Changes
### Before
n-news-theme-blocks % npm run lint

> fusion-news-theme-blocks@ lint /Users/howardj/sites/fusion-news-theme-blocks
> eslint --ext js --ext jsx blocks components --no-error-on-unmatched-pattern


/Users/howardj/sites/fusion-news-theme-blocks/blocks/ads-block/features/ads/default.jsx
  54:6  warning  React Hook useCallback has missing dependencies: 'debug', 'propsWithContext', and 'siteVars'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Users/howardj/sites/fusion-news-theme-blocks/blocks/header-nav-block/features/navigation/_children/search-box.jsx
  17:23  warning  Assignments to the 'disabledBtn' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect  react-hooks/exhaustive-deps

/Users/howardj/sites/fusion-news-theme-blocks/blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/search-box.jsx
  17:23  warning  Assignments to the 'disabledBtn' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect  react-hooks/exhaustive-deps

✖ 3 problems (0 errors, 3 warnings)

### After

howardj@it-20750 fusion-news-theme-blocks % npm run lint

> fusion-news-theme-blocks@ lint /Users/howardj/sites/fusion-news-theme-blocks
> eslint --ext js --ext jsx blocks components --no-error-on-unmatched-pattern

howardj@it-20750 fusion-news-theme-blocks % 


working as expected running locally still
![Screen Shot 2020-12-10 at 14 19 29](https://user-images.githubusercontent.com/5950956/101783876-b221fa80-3af2-11eb-8f00-36a2a190aecb.png)
![Screen Shot 2020-12-10 at 14 18 32](https://user-images.githubusercontent.com/5950956/101783879-b3532780-3af2-11eb-8d44-f8bbfd04bea3.png)
![Screen Shot 2020-12-10 at 14 18 28](https://user-images.githubusercontent.com/5950956/101783881-b3ebbe00-3af2-11eb-85a6-2eac3a799a69.png)


## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x ] Confirmed all the test steps a reviewer will follow above are working. 
- [ x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [ x] Ran `npm run test`, made sure all tests are passing
  - [x ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
